### PR TITLE
add content for issue 994

### PR DESCRIPTION
**Pull request by Bradamant3**
Signed-off-by: JENNIFER RONDEAU <jrondeau@heptio.com>

addresses issue #994 

note that we'll revisit entire approach when we redo install docs (fast follow after 0.10 release)

### CODE REVIEW

File: docs/aws-config.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ To set up Ark on AWS, you:
 * Configure the server
 * Create a Secret for your credentials
 
-If you do not have the `aws` CLI locally installed, follow the [user guide][5] to set it up.
+If you do not have the AWS CLI locally installed, follow the [user guide][5] to set it up.
```
Comment (nrb) [inline docs/aws-config.md]: I think I prefer the previous version, or maybe "AWS CLI client".

File: docs/install-overview.md [hunk 2]
```diff
@@ -6,6 +6,16 @@ In version 0.7.0 and later, you can run Ark in any namespace, which requires add
 
 In version 0.9.0 and later, you can use Ark's integration with restic, which requires additional setup. See [restic instructions][20].
 
+## Customize configuration
+
+Whether you run Ark on a cloud provider or on-premises, you can specify the name of the default volume snapshot when you set up the server:
+
+```bash
+kubectl apply -f examples/common/00-prereqs.yaml --default-volume-snapshot-locations <location-1>,<location-2>,<location-3>,...
```
Comment (nrb) [inline docs/install-overview.md]: The command here is `ark server`, not `kubectl apply`.

So `ark server --default-volume-snapshot-locations gcp:gcp-default`, which is an example from testing on my laptop.

File: docs/aws-config.md (hunk 1 above)
Comment (Bradamant3) [reply docs/aws-config.md]: I'll revert for now and we can discuss later. Bikeshedding not needed :-)

File: docs/install-overview.md [hunk 3]
```diff
@@ -6,6 +6,16 @@ In version 0.7.0 and later, you can run Ark in any namespace, which requires add
 
 In version 0.9.0 and later, you can use Ark's integration with restic, which requires additional setup. See [restic instructions][20].
 
+## Customize configuration
+
+Whether you run Ark on a cloud provider or on-premises, you can specify the name of the default volume snapshot when you set up the server by running this additional command:
```
Comment (skriss) [inline docs/install-overview.md]: I would say "...on-premises, if you have more than one volume snapshot location for a given PV provider, you can specify the one to be used by default when taking backups by setting a server flag in your Ark deployment YAML:"

File: docs/install-overview.md [hunk 4]
```diff
@@ -6,6 +6,16 @@ In version 0.7.0 and later, you can run Ark in any namespace, which requires add
...
+
+```bash
+ark server --default-volume-snapshot-locations <location-1>,<location-2>,<location-3>,...
```
Comment (skriss) [inline docs/install-overview.md]: This isn't a command the user would run, they would amend their Ark deployment YAML to include this flag.

Also, the value of the flag should be formatted as `provider-1:location-1,provider-2,location-2,...`

File: docs/install-overview.md [hunk 5]
```diff
@@ -6,6 +6,16 @@ In version 0.7.0 and later, you can run Ark in any namespace, which requires add
...
+```
+
+where `<location-1>,<location-2>,<location-3>,...` = one or more of your environment's snapshot locations, comma-delimited.
```
Comment (skriss) [inline docs/install-overview.md]: see prev comment about proper format

File: docs/install-overview.md (hunk 4 above)
Comment (Bradamant3) [reply docs/install-overview.md]: lol this is what I originally thought. Overthinkers r us.

**Comment by Bradamant3** [comment]
Took a slightly different approach with latest commit. In addition to TODO marked in one file, still need to add YAML for azure, gcp, ibm. Does not account for on-prem, not sure where to put bc we don't have a clear story there yet.

File: docs/aws-config.md (hunk 1 above)
Comment (skriss) [reply docs/aws-config.md]: +1 to @nrb's comment

File: docs/aws-config.md [hunk 6]
```diff
@@ -147,6 +147,10 @@ Specify the following values in the example files:
 
   * Replace `<YOUR_REGION>`. See the [VolumeSnapshotLocation definition][6] for details.
 
+* (Optional, use only if you need to specify multiple volume snapshot locations) In `examples/aws/10-deployment.yaml` (or `examples/aws/10-deployment-kube2iam.yaml`, as appropriate):
```
Comment (skriss) [inline docs/aws-config.md]: s/need to specify/have

File: docs/install-overview.md [hunk 7]
```diff
@@ -6,6 +6,12 @@ In version 0.7.0 and later, you can run Ark in any namespace, which requires add
 
 In version 0.9.0 and later, you can use Ark's integration with restic, which requires additional setup. See [restic instructions][20].
 
+## Customize configuration
+
+Whether you run Ark on a cloud provider or on-premises, if you have more than one volume snapshot location for a given volume provider, you can specify the default provider for backups by setting a server flag in your Ark deployment YAML. See TODO XREFS TO COME. DOES NOT ADDRESS ON-PREM.
```
Comment (skriss) [inline docs/install-overview.md]: 👍

File: docs/aws-config.md (hunk 6 above)
Comment (Bradamant3) [reply docs/aws-config.md]: or omit altogether ("use to specify multiple ...). Will fix up when the rest is done.

File: docs/install-overview.md (hunk 5 above)
Comment (Bradamant3) [reply docs/install-overview.md]: whoops, omitted to delete earlier :-(

File: docs/azure-config.md [hunk 8]
```diff
@@ -133,6 +133,9 @@ Now that you have your Azure credentials stored in a Secret, you need to replace
 
   * Replace `<YOUR_TIMEOUT>`. See the [VolumeSnapshotLocation definition][8] for details.
 
+* (Optional, use only if you need to specify multiple volume snapshot locations) In `examples/aws/10-deployment.yaml` (or `examples/aws/10-deployment-kube2iam.yaml`, as appropriate):
```
Comment (nrb) [inline docs/azure-config.md]:
```suggestion
* (Optional, use only if you need to specify multiple volume snapshot locations) In `examples/azure/10-deployment.yaml`:
```

File: docs/gcp-config.md [hunk 9]
```diff
@@ -119,6 +119,10 @@ Specify the following values in the example files:
 
     * Replace `<YOUR_STORAGE_CLASS_NAME>` with `standard`. This is GCP's default `StorageClass` name.
 
+* (Optional, use only if you need to specify multiple volume snapshot locations) In `examples/aws/10-deployment.yaml` (or `examples/aws/10-deployment-kube2iam.yaml`, as appropriate):
```
Comment (nrb) [inline docs/gcp-config.md]:
```suggestion
* (Optional, use only if you need to specify multiple volume snapshot locations) In `examples/gcp/10-deployment.yaml`:
```

File: docs/ibm-config.md [hunk 10]
```diff
@@ -57,7 +57,9 @@ Specify the following values in the example files:
 
   * Replace `<YOUR_BUCKET>`, `<YOUR_REGION>` and `<YOUR_URL_ACCESS_POINT>`. See the [BackupStorageLocation definition][6] for details.
 
+* (Optional, use only to specify multiple volume snapshot locations) In `examples/aws/10-deployment.yaml` (or `examples/aws/10-deployment-kube2iam.yaml`, as appropriate):
```
Comment (nrb) [inline docs/ibm-config.md]:
```suggestion
* (Optional, use only to specify multiple volume snapshot locations) In `examples/ibm/10-deployment.yaml`:
```

I'm not 100% sure this is relevant on IBM - their object storage is supported via the S3 API, but their own block storage isn't, unless the cluster has something like Ceph or Portworx.

File: docs/install-overview.md [hunk 11]
```diff
@@ -6,6 +6,12 @@ In version 0.7.0 and later, you can run Ark in any namespace, which requires add
 
 In version 0.9.0 and later, you can use Ark's integration with restic, which requires additional setup. See [restic instructions][20].
 
+## Customize configuration
+
+Whether you run Ark on a cloud provider or on-premises, if you have more than one volume snapshot location for a given volume provider, you can specify the default provider for backups by setting a server flag in your Ark deployment YAML.
```
Comment (skriss) [inline docs/install-overview.md]: "...for a given volume provider, you can specify its default location for backups..."

**Comment by Bradamant3** [comment]
@nrb what a very polite way to call out copy/paste errors :-) Thanks!

**Comment by Bradamant3** [comment]
Will squash to fix up signoff error

**Pull request merged**